### ISSUE TITLE
DataModule 도입하여 train /test loader 로 train & test 로드.

### ISSUE THREAD

**Issue by teang1995** [eubinecto/wisdomify#32]
### 1. 데이터 버전화 이유?
데이터가 일정 수준 이상 모이면 기존 데이터에 병합시켜 모델을 다시 학습시켜야  하기 때문. wisdomify 의 경우 인터넷에서 수집하거나, 라벨러가 직접 용례를 입력할 것이므로 데이터에 결함이 있을 가능성이 크다. 이런 데이터가 많아질 경우 데이터의 양이 늘어나도 모델의 성능이 떨어지는 경우가 있을 것. 이를 방지하고 효율적으로 데이터를 관리하기 위해 데이터를 버전화 한다.

### 2. DataModule?
현재 dataloader 의 load_wisdom2eg가 구현돼있지 않으므로 구현해주어야 함. 데이터가 늘어남에 따라  train/test split 을 해줄 수 있게 되었다. 이 때 pytorch의 datamodule을 사용하면 기존 코드에 비해 편리하게  dataloader  를 구성할 수 있어 이를 사용하고자 한다.
torch.utils.data 에서 random_split을 사용하면 train, valid, test set 으로 쉽게 나눌 수 있으니 이를 사용하여 분류하면 될 듯 하다.

괜찮은 예시를 찾아 [링크](https://www.askpython.com/python/pytorch-lightning)를 공유합니다-!

## To-do's
- [x] 기존 데이터를 wisdomify2eg_0.tsv, 추가된 데이터를 합친 파일을 wisdomify2eg_1.tsv로 명명한다.
- [ ] 추후 데이터를 입력받을 형식을 정하고, 데이터를 갱신할 수 있는 코드를 작성한다.

**Comment by eubinecto** [comment]
Why  &  To-do를 먼저 생각하고 채워주시면 좋을 것 같습니다! :)

**Comment by teang1995** [comment]
기존 파일을 wisdom2eg_0.tsv, 데이터를 추가한 파일을  wisdom2eg_1.tsv  로 저장했다. 데이터를 추가로 수집하기 전에 데이터 스키마를 지정해야 할 것 같다. 이것만 통일해주면 데이터를 합치고, 버전을 나누는 과정을 자동화할 수 있을 것!

**Comment by teang1995** [comment]
loader 구현 완료!  trainset, validset, test set을 80:10:10의 비율로 나누었다. 모듈 자체가 워낙 심플하다 보니 별로 손 댈 것도 없이.. 구현이 됐는데, 

데이터 합친 후 총 레코드 수가 984개이고, batch_size를 8로 주어 길이를  출력해보니 
`
len_total_data : 984
len_train_loader : 99
len_valid_loader : 13
len_test_loader : 13
`
위와 같이 잘 나오는 것을 확인할 수 있었음!

기존 코드에서  data loader를 구성하는 부분의 코드를 변경해줘야 한다. 본업하러 가야 하니 여기까지!

**Comment by teang1995** [comment]
VOCAB.py 에는 '산넘어 산', 수집된 데이터에는 '산 넘어 산' 으로 돼있어 에러가 발생했다. 현재 수정됨! 앞으로 용례 수집 시 유의해야 할 부분이다.

**Comment by teang1995** [comment]
그리고 방금 확인한 사실인데, egs_KoreaUniv_v0.tsv에는 '서당개 삼년이면 풍월을 읊는다'에 해당하는 용례가 수집되지 않았음! 내일 회의 때 얘기해보면 될 것 같다.

**Comment by teang1995** [comment]
pytorch lightning 의 Trainer에는 .train, .validate, .test 가 지원이 되는데, 이를 통해 매 에폭 검증 과정을 거치고, 학습 후 테스트를 해주면 되겠다. 다만 Trainer에서 max_epoch까지 정해주기 때문에 매 epoch 학습-검증의 단계를 밟게 하는 방법은 찾아봐야 할 것 같다. 

[pytorch-lightning 도큐먼트](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html) 에서 
<img width="821" alt="스크린샷 2021-08-06 오전 12 55 54" src="https://user-images.githubusercontent.com/23451721/128382116-04950da9-6924-4071-92fc-31443a34b24e.png">
와 같이 설명하고 있는데, 이해한 게 맞다면 이를 참고하면 해결 가능할 듯 하다.

**Comment by teang1995** [comment]
우선 torch-lightning의 datamodule로 구성한 loader 에서 학습이 되는 것을 확인했다. 
![스크린샷 2021-08-06 오전 2 43 17](https://user-images.githubusercontent.com/23451721/128396533-4161c50f-bc04-448f-87fa-6079cda7f6c3.png)
학습된 모델의 가중치 파일, 데이터 등의 버전 구분은 금요일 회의 때 결정될 것이니 이후 개발 시작하여 일요일까지 마무리하면 되겠다!
